x-pack/plugins/apm/server/routes/transactions/route.ts

```diff
@@ -364,10 +365,64 @@ const transactionChartsErrorRateRoute = createApmServerRoute({
   },
 });
 
+const transactionChartsColdstartRateRoute = createApmServerRoute({
+  endpoint:
+    'GET /internal/apm/services/{serviceName}/transactions/charts/coldstart_rate',
+  params: t.type({
+    path: t.type({
+      serviceName: t.string,
+    }),
+    query: t.intersection([
+      t.type({ transactionType: t.string }),
+      t.partial({ transactionName: t.string }),
```

Comment:
Personally I'd prefer to separate routes, one where transaction name is required, and one where it is not allowed. We've seen a couple of bugs where having optional parameters led to us forgetting to add them in the appropriate places.

Reply:
Added another route with required `transactionName`